cmd/tailscale/cli/up.go

```diff
@@ -552,9 +552,17 @@ func checkForAccidentalSettingReverts(flagSet map[string]bool, curPrefs *ipn.Pre
 	flagExplicitValue := map[string]interface{}{} // e.g. "accept-dns" => true (from flagSet)
 	for i := 0; i < prefType.NumField(); i++ {
 		prefName := prefType.Field(i).Name
+		// Persist is a legacy field used for storing keys, which
+		// probably should never have been part of Prefs. It's
+		// likely to migrate elsewhere eventually.
 		if prefName == "Persist" {
 			continue
 		}
+		// LoggedOut is a preference, but running the "up" command
```

Comment:
I wanted to write a test to prevent similar bugs like this in the future. I think the easiest place would be in tstest/integration/ (just have it do a sequence of up; logout; up --flags). Unfortunately I don't think the integration test is quite ready to execute `tailscale up` yet.

Reply:
cli_test.go has a huge table of these tests

Reply:
(it's not in up_test.go because the plan is to unify some code between "tailscale up" and the future "tailscale set")